src/Sarif.Viewer.VisualStudio.ResultSources.GitHubAdvancedSecurity.Core/Services/GitHubSourceService.cs

```diff
@@ -188,16 +188,20 @@ public async Task InitializeAsync()
         /// <inheritdoc cref="IGitHubSourceService.IsActiveAsync()"/>
         public async Task<Result> IsActiveAsync()
         {
-            if (this.GetOptionStateCallback != null && this.GetOptionStateCallback("GitHubAdvancedSecurity"))
+            if (this.GetOptionStateCallback != null)
             {
-                if (string.IsNullOrWhiteSpace(this.repoPath))
+                bool.TryParse(this.GetOptionStateCallback("GitHubAdvancedSecurity").ToString(), out bool ghAsEnabled);
+                if (ghAsEnabled)
```

Comment:
You can collapse this into the if() above:

if (this.GetOptionStateCallback != null
&& bool.TryParse(this.GetOptionStateCallback("GitHubAdvancedSecurity").ToString(), out bool ghAsEnabled)
&& ghAsEnabled)
{
...

Reply:
Did not know I could use out variables in the same line, good to know :)